Lib/dataset.py

```diff
@@ -1789,8 +1792,8 @@ def createVariableCopy(self, var, id=None, attributes=None, axes=None, extbounds
                         attributes['_FillValue'] = numpy.array(var.missing_value).astype(var.dtype)
                         attributes['missing_value'] = numpy.array(var.missing_value).astype(var.dtype)
                 else:
-                    attributes['_FillValue'] = fill_value
-                    attributes['missing_value'] = fill_value
+                    attributes['_FillValue'] = numpy.array(fill_value).astype(var.dtype)
```

Comment:
hum... did you test that with 32bit? I remember numpy complaining a lot about float32 casting